src/Runtime/StandardCallInterceptor.php

```diff
@@ -0,0 +1,195 @@
+        }
+
+        return (string) $payload[$aggregateIdPayloadKey];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function callContextProvider($contextProvider, Message $command)
+    {
+        if (! $contextProvider instanceof ContextProvider) {
+            throw new RuntimeException(
+                'By default a ContextProvider should implement the interface: '
+                . ContextProvider::class . '. Got ' . ((\is_object($contextProvider) ? \get_class($contextProvider) : \gettype($contextProvider)))
+            );
+        }
+
+        return $contextProvider->provide($command);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function callFirstAggregateFunction(string $aggregateType, callable $aggregateFunction, Message $command, $context = null): \Generator
+    {
+        $events = $aggregateFunction($command, $context);
+
+        if (! $events instanceof \Generator) {
+            throw NoGeneratorException::forAggregateTypeAndCommand($aggregateType, $command);
+        }
+
+        yield from new MapIterator($events, function ($event) use ($aggregateType, $command) {
+            if (null === $event) {
+                return null;
+            }
+
+            return $this->mapToMessage($event, $aggregateType, $command);
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function callSubsequentAggregateFunction(string $aggregateType, callable $aggregateFunction, $aggregateState, Message $command, $context = null): \Generator
+    {
+        $events = $aggregateFunction($aggregateState, $command, $context);
+
+        if (! $events instanceof \Generator) {
+            throw NoGeneratorException::forAggregateTypeAndCommand($aggregateType, $command);
+        }
+
+        yield from new MapIterator($events, function ($event) use ($aggregateType, $command) {
+            if (null === $event) {
+                return null;
+            }
+
+            return $this->mapToMessage($event, $aggregateType, $command);
+        });
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function callApplyFirstEvent(callable $applyFunction, Message $event)
+    {
+        return $applyFunction($event);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function callApplySubsequentEvent(callable $applyFunction, $aggregateState, Message $event)
+    {
+        return $applyFunction($aggregateState, $event);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function prepareNetworkTransmission(Message $message): Message
+    {
+        return $message;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function convertMessageReceivedFromNetwork(Message $message, $receivedFromEventStore = false): Message
+    {
+        return $message;
+    }
+
+    private function mapToMessage($event, string $aggregateType, Message $command): Message
+    {
+        if (! \is_array($event) || ! \array_key_exists(0, $event) || ! \array_key_exists(1, $event)
```

Comment:
Instead of using `\is_array` we should [import it](https://secure.php.net/manual/en/language.namespaces.importing.php) `use function is_array;` and so on.

Reply:
please not. I really prefer it this way so you easily see that it is a function available in global namespace.

Reply:
is this defined in a PSR?

Reply:
No, it comes from [Zend Coding Standard](https://github.com/zendframework/zend-coding-standard/pull/1) (seen in Zend\Diactoros). It was only a suggestion.

Reply:
ok, if it is not defined in a PSR I vote against it. It looks a bit nicer, but I see it like @prolic All the additional imports at the top of the file for no real benefit. And especially for exceptions it is very useful to see directly if a PHP standard exception is thrown or a project or namespace specific exception is used.

PHPStorm does not import global namespace by default, too. I think that's configurable, but it's not the default.

Reply:
I don't think it would be slower if you imported the function using `use function is_array;`. To my knowledge the reason it is slower is because PHP has to look in the local namespace and afterwards in the global namespace for a given function if you omit the backslash. But this should not be the case if you explicitly import the function into the local namespace and it should therefore not matter performance wise. That said I also do not like importing the functions either and the backslash makes it pretty clear that you're referring to something in the global namespace